Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -833,23 +832,22 @@ def network_get_state(self, usercallback, getpeerlist, sessioncalling=False):
                                    seeding_stats=seeding_stats, filepieceranges=self.filepieceranges, logmsgs=logmsgs)
                 self.progressbeforestop = ds.get_progress()
 
-            if sessioncalling:
+            if usercallback:
+                # Invoke the usercallback function via a new thread.
+                # After the callback is invoked, the return values will be passed to the
+                # returncallback for post-callback processing.
+                if not self.done:
+                    # runs on the reactor
+                    def session_getstate_usercallback_target():
+                        when, getpeerlist = usercallback(ds)
+                        if when > 0.0:
+                            # Schedule next invocation, either on general or DL specific
+                            self.session.lm.rawserver.add_task(lambda: self.network_get_state(usercallback, getpeerlist), when)
+
+                    callInThreadPool(session_getstate_usercallback_target)
```

Comment:
Why did you choose to go around the new rawserver, and call callInThreadPool directly?